packages/dev/core/src/scene.ts

```diff
@@ -3025,8 +3025,8 @@ export class Scene extends AbstractScene implements IAnimatable, IClipPlanesHold
      * @param allowMultiMaterials determines whether multimaterials should be considered
      * @returns the material or null if none found.
      */
-    public getMaterialByName(name: string, allowMultiMaterials: boolean = false): Nullable<Material> {
-        return this._getMaterial(allowMultiMaterials, (m) => m.name === name);
+    public getMaterialByName<T extends Material | MultiMaterial = Material>(name: string, allowMultiMaterials: boolean = false): Nullable<T> {

```

Comment:
MultiMaterial already extends Material, why was it necessary to specify it here?